lib/mutant/mutation.rb

```diff
@@ -40,11 +40,11 @@ def original_source
 
     # Test if mutation is killed by test reports
     #
-    # @param [Array<Report::Test>] test_reports
+    # @param [Result::Test] test_result
     #
     # @return [Boolean]
-    def self.success?(test_result)
-      self::TEST_PASS_SUCCESS.equal?(test_result.passed)
+    def success?(test_result)
```

Comment:
> Also I wasn't factoring in the constant at all. That isn't relevant imho. My primary consideration is if actual instance state is used or not. If it's not, then the method should be a singleton method.

As a general guideline, yes. But there are great examples to derivate from the general rule.

I care more about the clean-ness of the public API than the clean-ness of the implementation of the public API!

Hence I prefer a call side: `mutation.success?(test_result)` over `mutant.class.success?(test_result)`.

Since having to remember (and document / test) the 2nd one is a nightmare.

From the call sides POV I want to know if "this test result kills that specific mutation" and I'm not at all interested in the detail if for all instances of that mutation class the "killed mutation" decision is done the same.

Also as I noted I've a private branch that uses instance specific state to determine the kill decsion, one that depends on this change.

There are many good examples "algebraic OO components" as they are building an algebra need to violate the "has to use instance state rule".

Cases like:

```ruby
def success?(_input)
  false
end
```

Are well justified when part of an LSP compatible class hierarchy where "dynamically (with instance state) computated" versions exit.

Examples of the "keep the instance API well formed, but delegate to singleton for deduplication" are to be found in another popular library that implements something more complex than this cases small "mutant result algebra":

* https://github.com/dkubb/axiom/blob/master/lib/axiom/aggregate.rb#L36-L46